functions/src/quiz/round-end.ts

```diff
@@ -21,39 +21,48 @@ export const QuizRoundEnd = functions
             throw new HttpsError('permission-denied', 'User is not a game host');
         }
 
+        const playerRef = gameRef
+            .collection('players');
+
+        const players = await playerRef.get();
+
         const quizRef = admin.firestore()
             .collection('quizzes')
             .doc(data.quizId);
 
         const quizData = (await quizRef.get()).data();
 
         if (!quizData) {
-            throw new HttpsError('not-found', 'Not Found');
+            throw new HttpsError('not-found', 'No Quiz Found');
         }
 
         const roundRef = quizRef
             .collection('rounds')
             .doc(data.roundId);
 
-        const questions = await roundRef
-            .collection('questions')
-            .listDocuments();
+        const round = (await roundRef.get()).data();
+
+        // Build leaderboard
+        const leaderboard = players.docs.reduce((leaders, doc) => {
+            const player = doc.data();
+            leaders[ player.user ] = { answers: Array(round.questionList.length).fill(null), score: 0 };
+            return leaders;
+        }, <any> {});
```

Comment:
`any`?

Reply:
yes, need to manipulate the object and there is no starting type.... feel free to add some type models sometime then can change this :p